sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1563,10 +1563,12 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
     new AnalysisException(s"'$operation' does not support partitioning")
   }
 
-  def mixedRefsInAggFunc(funcStr: String): Throwable = {
-    val msg = "Found an aggregate function in a correlated predicate that has both " +
-      "outer and local references, which is not supported: " + funcStr
-    new AnalysisException(msg)
+  def mixedRefsInAggFunc(funcStr: String, origin: Origin): Throwable = {
+    new AnalysisException(
+      errorClass = "UNSUPPORTED_SUBQUERY_EXPRESSION_CATEGORY",
+      errorSubClass = "AGGREGATE_FUNCTION_MIXED_OUTER_LOCAL_REFERENCES",
+      origin = origin,
+      messageParameters = Map("function" -> funcStr))
```

Comment:
Should we quote the function? by `toSQLExpr` if it is an expression or maybe by `toSQLId`? cc @srielau

Reply:
Good Q,
Looks like this is a `sql` string generated from a predicate which may contain multiple functions. It renders from the error-classes.json file like this:

```
Found an aggregate function in a correlated predicate that has both outer and local references, which is not supported: <function>
```

It might be better to leave it unquoted since some parts of the predicate may contain their own double-quotes, e.g. literal string values. Will leave it alone for now, Serge please comment if you want it to work differently and I can update it.